client/layout/navigation.js

```diff
@@ -22,15 +24,24 @@ import { isWCAdmin } from '../dashboard/utils';
 import { addHistoryListener } from '../navigation/utils';
 
 const NavigationPlugin = () => {
-	const [ persistedQuery, setPersistedQuery ] = useState(
-		getPersistedQuery()
-	);
+	const { persistedQuery } = useSelect( ( select ) => {
+		return {
+			persistedQuery: select(
+				NAVIGATION_STORE_NAME
+			).getPersistedQueryFromStore(),
```

Comment:
Can we go ahead and use this same persisted queries in other components?  Or did you want to do a follow-up PR for this?

Reply:
I've been looking at this a bit, although there are ~22 files that reference persisted queries and I'm a little uncertain about switching everything over before I vet this revised approach. Let me know what you think.